firmware/components/driver_bus_i2c/driver_i2c.c

```diff
@@ -12,288 +12,287 @@
+esp_err_t driver_i2c_init(void) {
+  static bool driver_i2c_init_done = false;
+  if (driver_i2c_init_done)
+    return ESP_OK;
+  ESP_LOGD(TAG, "init called");
+
+  // create mutex for I2C bus
+  driver_i2c_mux = xSemaphoreCreateMutex();
+  if (driver_i2c_mux == NULL)
+    return ESP_ERR_NO_MEM;
+
+  // configure I2C
+  i2c_config_t conf = {
+      .mode       = I2C_MODE_MASTER,
+      .sda_io_num = CONFIG_PIN_NUM_I2C_DATA,
 #ifdef CONFIG_PIN_PULL_I2C_DATA
-		.sda_pullup_en    = GPIO_PULLUP_ENABLE,
+      .sda_pullup_en = GPIO_PULLUP_ENABLE,
 #else
-		.sda_pullup_en    = GPIO_PULLUP_DISABLE,
+      .sda_pullup_en = GPIO_PULLUP_DISABLE,
 #endif
-		.scl_io_num       = CONFIG_PIN_NUM_I2C_CLK,
+      .scl_io_num = CONFIG_PIN_NUM_I2C_CLK,
 #ifdef CONFIG_PIN_PULL_I2C_CLK
-		.scl_pullup_en    = GPIO_PULLUP_ENABLE,
+      .scl_pullup_en = GPIO_PULLUP_ENABLE,
 #else
-		.scl_pullup_en    = GPIO_PULLUP_DISABLE,
+      .scl_pullup_en = GPIO_PULLUP_DISABLE,
 #endif
-		.master.clk_speed = CONFIG_I2C_MASTER_FREQ_HZ,
-	};
-	esp_err_t res = i2c_param_config(I2C_MASTER_NUM, &conf);
-	if (res != ESP_OK) return res;
-
-	res = i2c_driver_install(I2C_MASTER_NUM, conf.mode, I2C_MASTER_RX_BUF_DISABLE, I2C_MASTER_TX_BUF_DISABLE, 0);
-	if (res != ESP_OK) return res;
-
-	driver_i2c_init_done = true;
-	ESP_LOGD(TAG, "init done");
-	return ESP_OK;
+      .master.clk_speed = CONFIG_I2C_MASTER_FREQ_HZ,
+  };
+  esp_err_t res = i2c_param_config(I2C_MASTER_NUM, &conf);
+  if (res != ESP_OK)
+    return res;
+
+  res = i2c_driver_install(I2C_MASTER_NUM, conf.mode, I2C_MASTER_RX_BUF_DISABLE,
+                           I2C_MASTER_TX_BUF_DISABLE, 0);
+  if (res != ESP_OK)
+    return res;
+
+  driver_i2c_init_done = true;
+  ESP_LOGD(TAG, "init done");
+  return ESP_OK;
 }
 
-esp_err_t driver_i2c_read_reg(uint8_t addr, uint8_t reg, uint8_t *value, size_t value_len)
-{
-	esp_err_t res;
-	if (xSemaphoreTake(driver_i2c_mux, portMAX_DELAY) != pdTRUE)
-		return ESP_ERR_TIMEOUT;
-
-	i2c_cmd_handle_t cmd = i2c_cmd_link_create();
-
-	res = i2c_master_start(cmd);
-	assert( res == ESP_OK );
-	res = i2c_master_write_byte(cmd, ( addr << 1 ) | WRITE_BIT, ACK_CHECK_EN);
-	assert( res == ESP_OK );
-	res = i2c_master_write_byte(cmd, reg, ACK_CHECK_EN);
-	assert( res == ESP_OK );
-
-	res = i2c_master_start(cmd);
-	assert( res == ESP_OK );
-	res = i2c_master_write_byte(cmd, ( addr << 1 ) | READ_BIT, ACK_CHECK_EN);
-	assert( res == ESP_OK );
-	if (value_len > 1)
-	{
-		res = i2c_master_read(cmd, value, value_len-1, ACK_VAL);
-		assert( res == ESP_OK );
-	}
-	res = i2c_master_read_byte(cmd, &value[value_len-1], NACK_VAL);
-	assert( res == ESP_OK );
-	res = i2c_master_stop(cmd);
-	assert( res == ESP_OK );
-
-	res = i2c_master_cmd_begin(I2C_MASTER_NUM, cmd, 1000 / portTICK_RATE_MS);
-	i2c_cmd_link_delete(cmd);
-
-	if (xSemaphoreGive(driver_i2c_mux) != pdTRUE)
-	{
-		ESP_LOGE(TAG, "xSemaphoreGive() did not return pdTRUE.");
-	}
-
-	return res;
+esp_err_t driver_i2c_read_reg(uint8_t addr,
```

Comment:
I'd rather have function definitions on one line if possible, makes things easier to read than splitting things too much.

Reply:
Yeah, I don't like having too much splitting here either. We should still decide on some column limit. 100-ish?

Reply:
The relevant options here are BinPackArguments and BinPackParameters, together with ColumnLimit.

With BinPack* set to true, args/params are either on one line or all on their own line. Without, they're just distributed so they don't go over the ColumnLimit. It seems that we want BinPack*: false and some reasonable limit.

Reply:
How do you like the current version?